bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -1348,16 +1349,28 @@ void drawSelected(int itemIndex, GC gc, Rectangle bounds, int state ) {
 						xx = x;
 						yy = parent.onBottom ? y -1 : y + 1;
 						ww = width;
-						hh = height;
+						hh = height - 1;
 						if (!parent.single && !parent.simple) ww += curveWidth - curveIndent;
 						drawBackground(gc, shape, xx, yy, ww, hh, defaultBackground, image, colors, percents, vertical);
 					}
 				}
 
-				//Highlight MUST be drawn before the outline so that outline can cover it in the right spots (start of swoop)
+				//Highlight colors MUST be drawn before the outline so that outline can cover it in the right spots (start of swoop)
 				//otherwise the curve looks jagged
 				drawHighlight(gc, bounds, state, rightEdge);
 
+				// draw highlight marker of selected tab
+				Color previousColor = gc.getBackground();
+				gc.setBackground(item.getDisplay().getSystemColor(parent.shouldHighlight() ? SWT.COLOR_LIST_SELECTION : SWT.COLOR_WIDGET_DISABLED_FOREGROUND));
+				double widthRatio = parent.shouldHighlight() ? 0.8 : 0.25;
```

Comment:
Interesting. I like the change in the width, gives nice "animation" effect.
I'm still undecided what is the best width here. With `1 : 0.2` I see "OK" results on Windows and Linux, interestingly "1" doesn't mean, the line is full tab width.

Reply:
OK, let's go for 1 : 0.2 then.